src/app/core/msg-io.service.ts

```diff
@@ -163,11 +167,13 @@ export class MsgIoService {
         isRead: false,
         year: MsgIoService.currentYear,
         fromName: await this.authService.displayName(),
-        fromPic: await this.authService.profilePictureLink()
+        fromPic: await this.authService.profilePictureLink(),
+        key
       };
 
       // Send the message
-      await this.messagesCollection.add(message).catch(error => {
+      const msgRef = this.messagesCollection.doc(this.authService.uid() + '-' + message.to);
+      await msgRef.set(message).catch(error => {
```

Comment:
Just to let you know, try/catch blocks can be used on promises in async functions. But this line will still work, so no need to change it.

Reply:
Is there anywhere where they can't be used?

Reply:
What do you mean by "they". Try/catch blocks in async functions? If yes, well they can be used with "normal" synchronous code, and with promises called using an `await`. However, if for some reason the promise has not been called using an `await`, well then you need to use `.catch()`